# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PWindowGlobal.ipdl
# Commit: 09fcee21b246
# Full Hash: 09fcee21b246dbd3622203d214234ee5116805f4
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-07-10 21:50:49
# Regressor Bug: 1561395
# File Overlap Count: 1
# Description:
#   Bug 1561395 - Move drawSnapshot API to WindowGlobalParent and allow specifying the whole viewport as a rect. r=mattwoodrow,nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35842
# ==============================================================================

diff -r 0873b1c9bca7 -r 09fcee21b246 dom/ipc/PWindowGlobal.ipdl
--- a/dom/ipc/PWindowGlobal.ipdl	Thu Jul 04 00:40:02 2019 +0000
+++ b/dom/ipc/PWindowGlobal.ipdl	Wed Jul 10 16:45:46 2019 +0000
@@ -11,6 +11,9 @@
 include DOMTypes;
 
 using JSWindowActorMessageKind from "mozilla/dom/JSWindowActor.h";
+using mozilla::gfx::IntRect from "mozilla/gfx/Rect.h";
+using moveonly mozilla::gfx::PaintFragment from "mozilla/gfx/CrossProcessPaint.h";
+using nscolor from "nsColor.h";
 
 namespace mozilla {
 namespace dom {
@@ -40,6 +43,8 @@
                               uint64_t aSwitchId)
       returns (nsresult rv, nullable PBrowserBridge bridge);
 
+  async DrawSnapshot(IntRect? aRect, float aScale, nscolor aBackgroundColor) returns (PaintFragment retval);
+
   /**
    * Returns the serialized security info associated with this window.
    */